### ISSUE TITLE
Tracking issue: Multiprocessing pipe cleanup error

### ISSUE THREAD

**Issue by richardsheridan** [richardsheridan/trio-parallel#398]
A pipe FD can be invalid before it hits the multiprocessing `__del__` code on py311, macos. 
```
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/multiprocessing/connection.py", line 133, in __del__
    self._close()
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/multiprocessing/connection.py", line 377, in _close
    _close(self._handle)
OSError: [Errno 9] Bad file descriptor
```
So far the context is not very informative.

Examples: 

- https://github.com/richardsheridan/trio-parallel/actions/runs/7447187749/job/20259020134?pr=397#step:7:272

**Comment by richardsheridan** [comment]
Similar example on py311, windows:
https://github.com/richardsheridan/trio-parallel/actions/runs/8516839126/job/23326500069#step:7:195

different test, different OS, different deps... same python version though!

**Comment by richardsheridan** [comment]
my guess is that this is some kind of GC race that is implementation dependent, so its best to get rid of our [del trick](https://github.com/richardsheridan/trio-parallel/blob/24f8d687eb156001f49f535051a86e9163c60df3/trio_parallel/_proc.py#L151-L158), probably by fixing #198.

**Comment by richardsheridan** [comment]
probably closed by #412.
